src/index.ts

```diff
@@ -98,16 +98,35 @@ function transformUrlIntoRegex(url: string) {
  * @param path Path to normalize.
  */
 function normalizePath(path: string) {
-  // In addition to transforming `{pathParam}` into `:pathParam` we also need to escape cases where a non-variabled
-  // colon is next to a variabled-colon because if we don't `path-to-regexp` won't be able to correct identify where the
-  // variable starts.
-  //
-  // For example if the URL is `/post/:param1::param2` we'll be escaping it to `/post/:param1\::param2`.
   return (
     path
-      .replace(/{(.*?)}/g, ':$1')
+      // This regex transforms `{pathParam}` into `:pathParam` so we can regex against it. We're
+      // also handling quirks here like if there's an optional proceeding or trailing curly bracket
+      // (`{{pathParam}` or `{pathParam}}`) as any unescaped curlys, which would be present in
+      // `:pathParam}`, will throw a regex exception.
+      .replace(/({?){(.*?)}(}?)/g, function (str, ...args) {
```

Comment:
This regex was changed from `{(.*?)}` to handle optional proceeding `{` and trailing `}`.

Reply:
I'm surprised you don't have to escape the curly braces. aren't they reserved characters in regexes?

Reply:
You're absolutely right and I'm not really sure how this works because it works in [Patterns](https://krillapps.com/patterns/) too:

![Screen Shot 2022-03-15 at 10 35 53 AM](https://user-images.githubusercontent.com/33762/158437809-2ade45e0-f613-4770-b273-7e2ee43d3b60.png)

Reply:
maybe there's a default flag enabled that makes this work